demos/http/http_demo_s3_download_multithreaded/http_demo_s3_download_multithreaded.c

```diff
@@ -408,25 +412,32 @@ static bool downloadS3ObjectFile( const char * pHost,
             {
                 queueOpStatus = retrieveHTTPResponse( responseQueue, &responseItem );
 
-                if( queueOpStatus == QUEUE_OP_SUCCESS )
+                if( queueOpStatus == QUEUE_OP_FAILURE )
                 {
-                    requestCount -= 1;
-
-                    LogInfo( ( "Main thread received HTTP response" ) );
-                    LogInfo( ( "Response Headers:\n%.*s",
-                               ( int32_t ) responseItem.response.headersLen,
-                               responseItem.response.pHeaders ) );
-                    LogInfo( ( "Response Status:\n%u", responseItem.response.statusCode ) );
-                    LogInfo( ( "Response Body:\n%.*s\n", ( int32_t ) responseItem.response.bodyLen,
-                               responseItem.response.pBody ) );
+                    returnStatus = false;
+                }
+                else if( queueOpStatus == QUEUE_OP_SUCCESS )
+                {
+                    if( responseItem.response.statusCode != 206 )
+                    {
+                        LogError( ( "Recieved repsonse with unexpected status code: %d", responseItem.response.statusCode ) );
+                        returnStatus = false;
+                    }
+                    else
+                    {
+                        requestCount -= 1;
+
+                        LogInfo( ( "Main thread received HTTP response" ) );
```

Comment:
I think we should always print the response received, even if the statusCode is not 206, for debugging ease. Can you move these logs to before `if( responseItem.response.statusCode != 206 )`